### PR TITLE
Bump sabre/dav to 4.2.1

### PR DESCRIPTION
**Pull request by come-nc**
Signed-off-by: Côme Chilliet <come.chilliet@nextcloud.com>

Needed for PHP 8.1 compatibility

### CODE REVIEW

**Review by CarlSchwan — APPROVED** [review]

File: .patches/sabre-dav/fix-backendinterface-phpdoc.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/lib/DAVACL/PrincipalBackend/BackendInterface.php b/lib/DAVACL/PrincipalBackend/BackendInterface.php
+index 72717a59bf..7140a92954 100644
+--- a/lib/DAVACL/PrincipalBackend/BackendInterface.php
++++ b/lib/DAVACL/PrincipalBackend/BackendInterface.php
+@@ -110,7 +110,7 @@ public function searchPrincipals($prefixPath, array $searchProperties, $test = '
+      * @param string $uri
+      * @param string $principalPrefix
+      *
+-     * @return string
++     * @return string|null
```
Comment (ChristophWurst) [inline .patches/sabre-dav/fix-backendinterface-phpdoc.patch:10]: included in 4.2.1, right? https://github.com/sabre-io/dav/pull/1371/commits/3c648d02a932db15c8f14bafe2b80df0bf8d80fe

**Review by ChristophWurst — CHANGES_REQUESTED** [review]

File: .patches/sabre-dav/fix-backendinterface-phpdoc.patch (hunk 1 above)
Comment (come-nc) [reply .patches/sabre-dav/fix-backendinterface-phpdoc.patch:10]: Yes, should I bump again?

Comment (ChristophWurst) [reply .patches/sabre-dav/fix-backendinterface-phpdoc.patch:10]: yes. you can just do it in this PR :)

Comment (come-nc) [reply .patches/sabre-dav/fix-backendinterface-phpdoc.patch]: Done

**Pull request merged**